### PR TITLE
fix: switch datasource option list contains similar ds

### LINKED ISSUE #13326: [Bug]-[1500]:Switch Datasources menu in QueryEditor
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we open datasources menu in query editor the datasources which are not compatible shows up.

### Steps To Reproduce

- Create datasource airtable datasource
- Create postgres datasource
- To create query on postgres datasource open query editor
- In the switch datasource dropdown you will also see airtable datasource.
Screenshot 2022-04-26 at 6 38 47 PM

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

Switching Datasource on the query pane or api pane editor was showing non-similar datasources in dropdown. The list is now filtered using the `pluginId` of the current datasource with the other active datasources.

Fixes #13326

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Create an application
2. Add multiple datasources (some similar, some different)
3. Create a query or api for one of the above datasources
4. When the editor opens, go to switch datasource dropdown and click on it
5. You will now see a list of only similar datasources (earlier all the active datasources were listed)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Irongade — DISMISSED** [review]

**Review by Irongade — DISMISSED** [review]

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js [hunk 1]
```diff
@@ -77,6 +99,11 @@ describe("Switch datasource", function() {
     );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js:99]: Where is the query run on Ds - postgresDatasourceNameSecond ?

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js [hunk 2]
```diff
@@ -77,6 +99,11 @@ describe("Switch datasource", function() {
...
   });
 
+  it("4. Confirm mongo datasource is not present in the switch datasources dropdown", function() {
+    cy.get(".t--switch-datasource").click();
+    cy.get(".t--datasource-option").should("not.have", mongoDatasourceName);
+  });
+
   it("4. Delete the query and datasources", function() {
     cy.deleteQueryUsingContext();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js:109]: DS - 'postgresDatasourceNameSecond' is not deleted?
Also please correct the test case numbers.

**Review by Aishwarya-U-R — CHANGES_REQUESTED** [review]

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js:99]: done

Comment (AmanAgarwal041) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/QueryPane/SwitchDatasource_spec.js:104]: done

**Pull request merged**